src/json_struct.rs

```diff
@@ -4,178 +4,421 @@ use std::collections::btree_map::Iter as MapIter;
-                        stack.push((MapIter(keys.iter()), namebuf.len()));
+                }
+                VecIterSerde(mut iter) => {
+                    let (index, json) = match iter.next() {
+                        Some((index, json)) => (index, json),
+                        None => continue,
+                    };
+                    stack.push((VecIterSerde(iter), off));
+                    match *json {
+                        Value::I64(value) => {
+                            try!(writeln!(out, "{}[{}] = {}",
+                                namebuf, index, value));
+                        }
+                        Value::U64(value) => {
+                            try!(writeln!(out, "{}[{}] = {}",
+                                namebuf, index, value));
+                        }
+                        Value::F64(value) => {
+                            try!(writeln!(out, "{}[{}] = {}",
+                                namebuf, index, value));
+                        }
+                        Value::String(ref value) => {
+                            try!(writeln!(out, "{}[{}] = {:?}",
+                                namebuf, index, value));
+                        }
+                        Value::Bool(value) => {
+                            try!(writeln!(out, "{}[{}] = {}",
+                                namebuf, index, value));
+                        }
+                        Value::Null => {
+                            try!(writeln!(out, "{}[{}] = null",
+                                namebuf, index));
+                        }
+                        Value::Array(ref vec) => {
+                            use std::fmt::Write;
+                            write!(&mut namebuf, "[{}]", index).unwrap();
+                            try!(writeln!(out, "{} = []", namebuf));
+                            stack.push((
+                                VecIterSerde(vec.iter().enumerate()),
+                                namebuf.len()));
+                        }
+                        Value::Object(ref keys) => {
+                            use std::fmt::Write;
+                            write!(&mut namebuf, "[{}]", index).unwrap();
+                            try!(writeln!(out, "{} = {{}}", namebuf));
+                            stack.push((MapIterSerde(keys.iter()), namebuf.len()));
+                        }
                     }
                 }
+                _ => unreachable!(),
             }
         }
+        Ok(())
     }
-    Ok(())
 }
 
-
 #[cfg(test)]
-mod test {
-    use rustc_serialize::json::Json;
+mod test_serde {
+    use serde_json::value::Value;
+    use serde_json::de;
 
     use super::json_to_gron;
 
     fn assert_equal(json_src: &str, gron: &str) {
         let mut buf = Vec::new();
         json_to_gron(&mut buf, "json",
-            &Json::from_str(json_src).unwrap()).unwrap();
+            &de::from_str(json_src).unwrap() as &Value).unwrap();
```

Comment:
Sounds fine.

I don't care about commits too much. I'll probably use "Squash and merge" button on github :)